spec/models/miq_expression_spec.rb

```diff
@@ -92,7 +92,7 @@
       exp1 = {"STARTS WITH" => {"field" => "Vm-name", "value" => "foo"}}
       exp2 = {"ENDS WITH" => {"field" => "Vm-name", "value" => "bar"}}
       sql, * = MiqExpression.new("OR" => [exp1, exp2]).to_sql
-      expect(sql).to eq("\"vms\".\"name\" LIKE 'foo%' OR \"vms\".\"name\" LIKE '%bar'")
+      expect(sql).to eq("(\"vms\".\"name\" LIKE 'foo%' OR \"vms\".\"name\" LIKE '%bar')")
```

Comment:
> To clarify, do we need to ensure all subclasses of Expression implement non-abstract methods? 

I used to think this was what `NotImplementedError` was for....would raising `"abstract method"` or similar be sufficient? Not sure the right way to do this anymore....

> Should we make sure the timezone specific components do the right thing with respect to timezones? 

We have tests for this on the higher `MiqExpression#to_sql` level, but I have noticed that we have lost some test coverage since I have un-DRY'd some of this code (the datetime 'normalization' used to be much more terse, so it looked like we had good coverage before). I will address that in this PR. I'm a bit undecided with regard to adding tests for specific components at this point. I can't write good unit tests for them because this stuff is so coupled to Arel. So while they are necessarily integrated I feel better testing at the `MiqExpression` level for now and treating these as private implementation details.

> Do we need to have tests that demonstrate the parsing of @pexp into the right subclasses?

As above, this feels like testing implementation to me, so I feel better working to increase confidence that `MiqExpression#to_sql` does the right thing for all use cases.